File: java/DataStructureBenchmark/src/structures/HashTable/HashTable.java
package structures.HashTable;

import java.util.HashMap;
import java.util.Map;

import structures.common.Items;

public class HashTable {
	private  final Map<String, Items> hashTable=new HashMap<>();
	
	
    public void insert(String key, String keyString, String value) {
    	hashTable.put(key, new Items(key, keyString, value));
    }

    public Items search(String key) {
        return hashTable.get(key);
    }
}

File: php/DataStructureBenchmark/structures/HashTable.php
<?php

declare(strict_types=1);

class HashTable
{
    private array $table = [];

    public function insert(string $key, string $keyString, string $value): void
    {
        $this->table[$key] = [
            'keyString' => $keyString,
            'value' => $value,
        ];
    }

    public function search(string $key): ?array
    {
        return $this->table[$key] ?? null;
    }

    public function sortByKeyString(): void
    {
        uasort($this->table, static function (array $a, array $b): int {
            return strcmp($a['keyString'], $b['keyString']);
        });
    }

    public function count(): int
    {
        return count($this->table);
    }
}